src/sdk/tests/integration/TokenUpdateTransactionIntegrationTests.cc

```diff
@@ -44,10 +47,437 @@ class TokenUpdateTransactionIntegrationTests : public BaseIntegrationTest
+                                            std::shared_ptr<PrivateKey> newKey,
+                                            std::shared_ptr<PrivateKey> signerKey,
+                                            TokenKeyValidation tokenKeyValidation)
+  {
+    TokenUpdateTransaction tx =
+      TokenUpdateTransaction().setTokenId(tokenId).setTokenVerificationMode(tokenKeyValidation);
+
+    switch (updateKeyType)
+    {
+      case UpdateKeyType::WIPE_KEY:
+        tx.setWipeKey(newKey);
+        break;
+      case UpdateKeyType::KYC_KEY:
+        tx.setKycKey(newKey);
+        break;
+      case UpdateKeyType::SUPPLY_KEY:
+        tx.setSupplyKey(newKey);
+        break;
+      case UpdateKeyType::FREEZE_KEY:
+        tx.setFreezeKey(newKey);
+        break;
+      case UpdateKeyType::FEE_SCHEDULE_KEY:
+        tx.setFeeScheduleKey(newKey);
+        break;
+      case UpdateKeyType::PAUSE_KEY:
+        tx.setPauseKey(newKey);
+        break;
+      case UpdateKeyType::METADATA_KEY:
+        tx.setMetadataKey(newKey);
+        break;
+      case UpdateKeyType::ADMIN_KEY:
+        tx.setAdminKey(newKey);
+        break;
+      case UpdateKeyType::LOWER_PRIVILEGE:
+        tx.setWipeKey(newKey)
+          .setKycKey(newKey)
+          .setSupplyKey(newKey)
+          .setFreezeKey(newKey)
+          .setFeeScheduleKey(newKey)
+          .setPauseKey(newKey)
+          .setMetadataKey(newKey);
+        break;
+      case UpdateKeyType::ALL:
+        tx.setWipeKey(newKey)
+          .setKycKey(newKey)
+          .setSupplyKey(newKey)
+          .setFreezeKey(newKey)
+          .setFeeScheduleKey(newKey)
+          .setPauseKey(newKey)
+          .setMetadataKey(newKey)
+          .setAdminKey(newKey);
+        break;
+      case UpdateKeyType::NONE:
+        break;
+    }
+    TransactionResponse txResponse;
+
+    if (updateKeyType == UpdateKeyType::ALL || updateKeyType == UpdateKeyType::ADMIN_KEY ||
+        tokenKeyValidation == TokenKeyValidation::FULL_VALIDATION)
+    {
+      txResponse = tx.freezeWith(&getTestClient()).sign(newKey).sign(signerKey).execute(getTestClient());
+    }
+    else
+    {
+      txResponse = tx.freezeWith(&getTestClient()).sign(signerKey).execute(getTestClient());
+    }
+
+    return txResponse;
+  }
+
 private:
   const std::vector<std::byte> mMetadata = { std::byte(0xAA), std::byte(0xAB), std::byte(0xAC), std::byte(0xAD) };
 };
 
+// HIP-540 tests
+
+//-----
+TEST_F(TokenUpdateTransactionIntegrationTests, TokenUpdateTransactionUpdateKeysToZeroKey)
+{
+  // Given
+  std::shared_ptr<PrivateKey> operatorKey;
+  ASSERT_NO_THROW(operatorKey =
+                    ED25519PrivateKey::fromString("bbd0894de0b4ecfa862e963825c5448d2d17f807a16869526bff29185747acdb"));
+
+  std::shared_ptr<PrivateKey> zeroKey = ED25519PrivateKey::fromString(ED25519PrivateKey::ZERO_KEY_STR);
+
+  // Make token immutable
+  TransactionResponse txResponse = createTokenWithKeysHelper(UpdateKeyType::ALL, operatorKey, operatorKey);
```

Comment:
Could you add `// When` and `// Then` comments in this test?

Reply:
Also this could all be combined into:
```
TokenId tokenId;
EXPECT_NO_THROW(tokenId = createTokenWithKeysHelper(UpdateKeyType::ALL, operatorKey, operatorKey).setValidateStatus(true).getReceipt(getTestClient()).mTokenId.value());
```
This would allow the macro to catch more potential throws. Can be combined in other tests as well.